packages/components/data-table/src/data-table.tsx

```diff
@@ -239,8 +239,16 @@ export type TDataTableProps<Row extends TRow = TRow> = {
 };
 
 const DataTable = <Row extends TRow = TRow>(props: TDataTableProps<Row>) => {
+  const { columns, isCondensed } = useDataTableManagerContext();
+  const isValueFromProvider = Boolean(columns && columns.length !== 0);
+  const columnsData = isValueFromProvider ? columns : props.columns;
+  const condensedValue =
+    isValueFromProvider && isCondensed !== undefined
+      ? isCondensed
+      : props.isCondensed;
```

Comment:
Would it be easier and also work if check for the props first and then for the context values?
```suggestion
  const dataTableManagerProvider = useDataTableManagerContext();
  const columnsData = props.columns.length > 0 ? props.columns : dataTableManagerProvider.columns;
  const condensedValue = props.isCondensed ?? dataTableManagerProvider.isCondensed;
```

Reply:
Yes I tried this and the other way too, it looks simpler but where it becomes tricky is we are trying to check for the existence of one boolean value or the other. In which case, the absence of it is also a boolean value. That's why I tied it to the column from the provider `isValueFromProvider` so when we see that value or not, we can do a much better check for the `condensedValue`